dev_docs/tutorials/data/search.mdx

```diff
@@ -0,0 +1,490 @@
+---
+id: kibDevTutorialDataSearchAndSessions
+slug: /kibana-dev-docs/tutorials/data/search-and-sessions
+title: Kibana data.search Services
+summary: Kibana Search Services 
+date: 2021-02-10
+tags: ['kibana', 'onboarding', 'dev', 'tutorials', 'search', 'sessions', 'search-sessions']
+---
+
+## Search Service
+
+### Low Level Search 
+
+Searching data stored in `Elasticsearch` can be done in various ways, for example using the `Elasticsearch` REST API or using an `Elasticsearch Client` for low level access.
+
+However, the recommended and easist way to search `Elasticsearch` is by using the low level search service. The service is exposed from the `data` plugin, and by using it, you don't only gain access to the data you stored, but also to additional capabilities, such as Custom Search Strategies, Asynchronous Search, Partial Results, Search Sessions and more.
+
+Here is a basic example for using the `data.search` service from a custom plugin:
+
+```ts
+import { CoreStart, Plugin } from 'kibana/public';
+import { DataPublicPluginStart, isCompleteResponse, isErrorResponse } from '???';
+
+export interface MyPluginStartDependencies {
+  data: DataPublicPluginStart;
+}
+
+export class MyPlugin implements Plugin {
+  public start(core: CoreStart, { data }: MyPluginStartDependencies) {
+    const query = {
+      filter: [{
+        match_all: {}
+      }],
+    };
+    const req = {
+      params: {
+        index: 'my-index-*',
+        body: {
+          query,
+          aggs: {},
+        },
+      }
+    };
+    data.search.search(req).subscribe({
+      next: (result) => {
+        if (isCompleteResponse(res)) {
+          // handle search result
+        } else if (isErrorResponse(res)) {
+          // handle error, this means that some results were returned, but the search has failed to complete.
+        } else {
+          // handle partial results. See LINK TO PARTIAL RESULTS for more information.
+        }
+      }, 
+      error: (e) => {
+        // handle error thrown, for example a server hangup
+      },
+    })
+  }
+}
+```
+
+Note: The `data` plugin contains services to help you generate the `query` and `aggs` portions, as well as managing indicies using the `data.indexPatterns` service. You can read more about them [here]().  
+
+<DocCallOut title="Server Side Search">
+  The `data.search` service is available on both server and client, with similar APIs.
+</DocCallOut>
+
+#### Error handling
+
+The `search` method can throw several types of errors. 
```

Comment:
Not sure what we mean by  `several types`. Can we list the types or link to them? 
if no, then I'd omit the "several types" part